logic/strategy/src/channel_finalizer.rs

```diff
@@ -0,0 +1,253 @@
+use async_lock::RwLock;
+use async_trait::async_trait;
+use chain_actions::channels::ChannelActions;
+use chain_db::traits::HoprCoreEthereumDbActions;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use hopr_internal_types::prelude::*;
+use hopr_platform::time::native::current_time;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+use std::ops::Sub;
+use std::sync::Arc;
+use std::time::Duration;
+use tracing::{debug, error, info};
+use validator::Validate;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+use hopr_metrics::metrics::SimpleCounter;
+
+use crate::strategy::SingularStrategy;
+use crate::{errors, Strategy};
+
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    static ref METRIC_COUNT_CLOSURE_FINALIZATIONS: SimpleCounter = SimpleCounter::new(
+        "hopr_strategy_closure_auto_finalization_count",
+        "Count of channels where closure finalizing was initiated automatically"
+    )
+    .unwrap();
+}
+
+/// Contains configuration of the [ClosureFinalizerStrategy].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, smart_default::SmartDefault, Validate, Serialize, Deserialize)]
+pub struct ClosureFinalizerStrategyConfig {
+    /// Do not attempt to finalize closure of channels that have
+    /// been overdue for closure for more than this period.
+    ///
+    /// Default is 3600 seconds.
+    #[default(Duration::from_secs(3600))]
+    pub max_closure_overdue: Duration,
```

Comment:
Channel closure finalization can happy at any time and by both parties. Why should we not finalize channels?

Reply:
60 minutes after the closure - that should be way too late.